test/no_only.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+var tap = require('tap');
+var path = require('path');
+var spawn = require('child_process').spawn;
+var concat = require('concat-stream');
+
+var stripFullStack = require('./common').stripFullStack;
+
+var tapeBin = path.join(process.cwd(), 'bin/tape');
+
+tap.test(
+    'Should throw error when --no-only is passed via cli and there is a .only test',
+    { todo: process.versions.node.match(/0\.10\.\d+/) ? 'Fails on node 0.10' : false },
```

Comment:
why does it fail on node 0.10?

Reply:
okay, so for node 0.10, the exit code was 8 instead of 1 for the cases where the error is thrown. 

I tried to look for similar use-cases in the tests and saw we try to skip such behaviour by excluding some versions, notable amongst them was https://github.com/substack/tape/blob/master/test/async-await.js.

Reply:
That test file skips in node < 8 because node 8 is what adds `async`/`await` syntax.

I can't find anywhere exiting with an `8`; I'll take a look at the failures and see what might be the problem.

Reply:
o okay makes sense. I installed 0.10 locally, removed the condition and run the tests and got the code as `8` for that as well hence my conclusion. I get the reason now though.